python/ray/util/sgd/utils.py

```diff
@@ -248,6 +251,31 @@ def check_for_failure(remote_values):
     return False
 
 
+class RayFileLock(FileLock):
+    """Utility for serializing data loading. Extends FileLock using a
+    temporary file as the lock file.
+
+    Example:
+        class CustomOperator(TrainingOperator):
+            def setup(self, config):
+                ...
+                # Load data
+                with RayFileLock():
+                    # All code here will be serialized across all workers.
+                    train_data = ...
```

Comment:
Please fix: 

I'd actually just prefer that we always lock to something obvious like `FileLock("~/.cache")`. 

Alternatively, if that's not preferable, let's instead do something like self.lock(filename="str.lock") in the operator. 

I'm quite opposed to having a utility wrapper around Filelock. It's a lot of overhead to remember how and where to import, etc.

Reply:
ok makes sense. I removed RayFileLock.